tempest/api/workloadmgr/multiattach_volumes/test_multiattach_volumes.py

```diff
@@ -0,0 +1,337 @@
+                reporting.set_test_script_status(tvaultconf.FAIL)
+
+            if md5sums_list[1] == md5sums_after_vol_selective1 or md5sums_list[1] == md5sums_after_vol_selective2:
+                LOG.debug("***MDSUMS MATCH***")
+                reporting.add_test_step(
+                    "Md5 Verification for volume disk for VM1", tvaultconf.PASS)
+            else:
+                LOG.debug("***MDSUMS DON'T MATCH*** expected: " + md5sums_list[1] + " actual: " + md5sums_after_vol_selective1)
+                reporting.add_test_step(
+                    "Md5 Verification for volume disk for VM1", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+
+            if md5sums_list[2] == md5sums_after_opt_selective2 or md5sums_list[2] ==md5sums_after_opt_selective1:
+                LOG.debug("***MDSUMS MATCH***")
+                reporting.add_test_step(
+                    "Md5 Verification for boot disk for VM2", tvaultconf.PASS)
+            else:
+                LOG.debug("***MDSUMS DON'T MATCH*** expected: " + md5sums_list[2] + " actual: " + md5sums_after_opt_selective2)
+                reporting.add_test_step(
+                    "Md5 Verification for boot disk for VM2", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+
+            if md5sums_list[3] == md5sums_after_vol_selective2 or md5sums_list[3] == md5sums_after_vol_selective1:
+                LOG.debug("***MDSUMS MATCH***")
+                reporting.add_test_step(
+                    "Md5 Verification for volume disk for VM2", tvaultconf.PASS)
+            else:
+                LOG.debug("***MDSUMS DON'T MATCH*** expected: " + md5sums_list[3] + " actual: " + md5sums_after_vol_selective2)
+                reporting.add_test_step(
+                    "Md5 Verification for volume disk for VM2", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+
+        else:
+            reporting.add_test_step("Selective restore of " + snapshot_type + " snapshot",
+                                    tvaultconf.FAIL)
+
+    @decorators.attr(type='workloadmgr_api')
+    def test_01_multiattach_volumes(self):
+        try:
+            test_var = "tempest.api.workloadmgr.multiattach_volumes.test_image_booted_"
+            tests = [[test_var + "workload_api", 0],
+                     [test_var + "full_snapshot_api", 0],
+                     [test_var + "incremental_snapshot_api", 0],
+                     # [test_var+"snapshot_mount_api", 0],
+                     # [test_var+"filesearch_api", 0],
+                     [test_var + "selectiverestore_api", 0],
+                     [test_var+"inplacerestore_api", 0]]
+                    # [test_var+"oneclickrestore_api", 0]]
+            reporting.add_test_script(tests[0][0])
+            self.kp = self.create_key_pair(tvaultconf.key_pair_name)
+            self.vm_id_1 = self.create_vm(key_pair=self.kp)
```

Comment:
You should have additional parameter "vm_cleanup=True" while creating vm instance. It will take care of deleting vm after tc execution. 
Also, in case of exception you should handle deletion of vm using self.delete_vm().

Same applies to vm_id_2 as well.

Reply:
in create_vm function the vm_cleanup is by default true